src/MOIWrapper.jl

```diff
@@ -0,0 +1,558 @@
+export ClpOptimizer
+
+using LinQuadOptInterface
+using Clp.ClpCInterface
+
+function replaceInf(x)
+    for i in 1:length(x)
+        if x[i] > 1e20
+            x[i] = Inf
+        elseif x[i] < -1e20
+            x[i] = -Inf
+        end
+    end
+    return x
+end
+
+function replaceInf(x, index)
+    if x[index] > 1e20
+        x[index] = Inf
+    elseif x[index] < -1e20
+        x[index] = -Inf
+    end
+end
+
+const LQOI = LinQuadOptInterface
+const MOI  = LQOI.MOI
+
+const SUPPORTED_OBJECTIVES = [
+    LQOI.Linear
+]
+
+const SUPPORTED_CONSTRAINTS = [
+    (LQOI.Linear, LQOI.EQ),
+    (LQOI.Linear, LQOI.LE),
+    (LQOI.Linear, LQOI.GE),
+    (LQOI.SinVar, LQOI.EQ),
+    (LQOI.SinVar, LQOI.LE),
+    (LQOI.SinVar, LQOI.GE),
+    (LQOI.SinVar, LQOI.IV),
+    (LQOI.VecVar, MOI.Nonnegatives),
+    (LQOI.VecVar, MOI.Nonpositives),
+    (LQOI.VecVar, MOI.Zeros),
+    (LQOI.VecLin, MOI.Nonnegatives),
+    (LQOI.VecLin, MOI.Nonpositives),
+    (LQOI.VecLin, MOI.Zeros)
+]
+
+mutable struct ClpOptimizer <: LQOI.LinQuadOptimizer
+    LQOI.@LinQuadOptimizerBase
+    # env
+    params::Dict{Symbol,Any}
+    ClpOptimizer(::Void) = new()
+end
+
+### Options
+
+# map option name to C function
+const optionmap = Dict(
+   :PrimalTolerance => set_primal_tolerance,
+   :DualTolerance => set_dual_tolerance,
+   :DualObjectiveLimit => set_dual_objective_limit,
+   :MaximumIterations => set_maximum_iterations,
+   :MaximumSeconds => set_maximum_seconds,
+   :LogLevel => set_log_level,
+   :Scaling => scaling,
+   :Perturbation => set_perturbation,
+   #:Algorithm => set_algorithm
+   )
+# These options are set by using the ClpSolve object
+const solveoptionmap = Dict(
+   :PresolveType => set_presolve_type,
+   :SolveType => set_solve_type,
+   :InfeasibleReturn => set_infeasible_return,
+   )
+
+function setoption(m::ClpOptimizer, name::Symbol, value)
+    if haskey(optionmap, name)
+        optionmap[name](m.inner,value)
+    elseif haskey(solveoptionmap, name)
+        solveoptionmap[name](m.solveroptions,value)
+    else
+        error("Unrecognized option: $name")
+    end
+end
+
+function ClpOptimizer(;kwargs...)
+    # env = Env()
```

Comment:
Style: don't leave code commented without an explanation why. If the code doesn't do anything, remove it.